test/runtests.jl

```diff
@@ -550,3 +550,30 @@ let
     @test srep[7] == 'β'
     @test_throws BoundsError srep[8]
 end
+
+
+## Reverse strings ##
+
+let
+    rs = RevString("foobar")
+    @test length(rs) == 6
+    @test sizeof(rs) == 6
+    @test isascii(rs)
+
+    # Base issue #4586
+    @test rsplit(RevString("ailuj"),'l') == ["ju","ia"]
+    @test parse(Float64,RevString("64")) === 46.0
+
+    # reverseind
+    for prefix in ("", "abcd", "\U0001d6a4\U0001d4c1", "\U0001d6a4\U0001d4c1c", " \U0001d6a4\U0001d4c1")
+        for suffix in ("", "abcde", "\U0001d4c1β\U0001d6a4", "\U0001d4c1β\U0001d6a4c", " \U0001d4c1β\U0001d6a4")
+            for c in ('X', 'δ', '\U0001d6a5')
+                s = convert(String, string(prefix, c, suffix))
+                s = RevString(s)
+                r = reverse(s)
+                ri = search(r, c)
```

Comment:
Should also check that `@test r == RevString(s)` (and not overwrite `s` with the reversed string).

Reply:
I hadn't really noticed how weird this code is...

Reply:
`r` isn't the same as `RevString(s)`, since `s` was redefined as the `RevString` of itself, so `reverse(s)` should then be the original `s` from before it was `RevString`'d.

Reply:
Yes but since the original `s` is lost, you can't check that the result is correct...

Reply:
These tests should be pretty much entirely rewritten.